crates/derive/src/traits/data_sources.rs

```diff
@@ -20,6 +20,18 @@ pub trait ChainProvider {
     async fn receipts_by_hash(&self, hash: B256) -> Result<Vec<Receipt>>;
 }
 
+/// Describes the functionality of a data source that fetches safe blocks.
+#[async_trait]
+pub trait SafeBlockFetcher {
+    /// Returns the block reference (info) given a block number.
+    /// Should error if the block does not exist.
+    async fn block_info_by_number(&self, number: u64) -> Result<BlockInfo>;
```

Comment:
Should be `L2BlockInfo`. Maybe rename to `l2_block_info_by_number` as well? `block_info_by_number` is already in `ChainProvider`